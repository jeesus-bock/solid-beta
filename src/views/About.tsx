import { useParams, Link } from '@solidjs/router';
import { createMemo, For, Suspense } from 'solid-js';
import { emptyContent } from '../api/contents';
import { Tag } from '../components/Tag';
import { TagList } from '../components/TagList';
import { useContentCtx } from '../context/contents';
import styles from './About.module.css';
import viewStyles from './View.module.css';
export default function About() {
  const contents = useContentCtx();
  if (!contents) {
    throw new Error('contents ctx data not available');
  }
  const data = createMemo(() => {
    if (!contents.loading) {
      const ret = contents.contents.get('about');
      if (ret) return ret;
    }
    return emptyContent;
  });
  return (
    <div class='flex'>
      <div style='display: flex;'>
        <Suspense fallback={<span>Suspended!</span>}>
          <div class={styles['links-bar']}>
            <Link href='/about/test1'>test1</Link>
            <Link href='/about/test2'>test2</Link>
            <Link href='/about/test3'>test3</Link>
            <Link href='/about/test4'>test4</Link>
          </div>
          <div class={viewStyles['intro']}>
            <h1>{data().title}</h1>
            <p>{data().desc}</p>

            <TagList tags={data().tags} />
          </div>
        </Suspense>
      </div>
      <div class={viewStyles['content']}>
        <p>{data().content}</p>
      </div>
    </div>
  );
}
