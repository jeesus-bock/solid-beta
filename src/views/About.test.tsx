import { describe, expect, test } from 'vitest';
import { render } from 'solid-testing-library';
// No named exports because of the router. TODO investigate.
import About from './About';

describe('views <About />', () => {
  test('Renders', async () => {
    const { container, unmount } = await render(() => <About />);
    expect(container).toMatchSnapshot();
    unmount();
  });
});
