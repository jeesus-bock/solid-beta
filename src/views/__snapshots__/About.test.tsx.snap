// Vitest Snapshot v1

exports[`<NavBar /> > Renders 1`] = `
<div>
  <div>
    About
  </div>
</div>
`;

exports[`views <About /> > Renders 1`] = `
<div>
  <div>
    About
  </div>
</div>
`;
